timely_beliefs/tests/test_belief_query.py

```diff
@@ -18,6 +18,36 @@
 from timely_beliefs.tests import session
 
 
+@pytest.fixture(scope="function")
+def belief_recorded_at_unique_knowledge_time(
+    unique_knowledge_time_sensor: DBSensor, test_source_a: DBBeliefSource
+):
+    """Define belief about a future event at its unique knowledge time (e.g. a publication date)."""
+    belief = DBTimedBelief(
+        source=test_source_a,
+        sensor=unique_knowledge_time_sensor,
+        value=10,
+        belief_time=datetime(1990, 5, 10, 0, tzinfo=utc),
+        event_start=datetime(1990, 6, 1, 0, tzinfo=utc),
+    )
+    session.add(belief)
+    return belief
+
+
+def test_query_belief_for_sensor_with_unique_knowledge_time(
+    unique_knowledge_time_sensor: DBSensor,
+    belief_recorded_at_unique_knowledge_time: DBTimedBelief,
+):
+    """Test query of sensor with a unique knowledge time, in combination with a belief time window."""
+    belief_df = DBTimedBelief.search_session(
+        session=session,
+        sensor=unique_knowledge_time_sensor,
+        beliefs_after=pd.Timestamp("1990-04-01 00:00Z"),
+        belief_before=pd.Timestamp("1990-06-01 00:00Z"),
+    ).convert_index_from_belief_time_to_horizon()
+    assert belief_df.belief_horizons[0] == timedelta(0)
```

Comment:
I'm confused what you're testing here. Why would the belief horizon of the event not be 21 days?

Reply:
Because the value was recorded exactly at the unique knowledge time of the sensor. The belief horizon is defined as the duration between belief time and knowledge time.

Reply:
I guess I could makes this clearer in the test by also including an assert statement regarding the knowledge horizon (which should be 22 days, from midnight May 10th to midnight June 1st).